rfcs/0000-light-rail-trip-model.md

```diff
@@ -0,0 +1,225 @@
+
+## Glides train - trip assignment events
+
+A new `com.mbta.ctd.glides.vehicle_trip_assignment` event will be established. The exact schema will be defined via the pull request review procedure on the [`mbta/schemas`](https://github.com/mbta/schemas/) repository, but there are some basic criteria that can be established. Events will at least have a `tripId` and `vehicleId` key. The `tripId` will be the scheduled or added trip ID from Glides, and the `vehicleId` will correspond to the ID of the vehicle in the GTFS-rt feed. Each event will represent the assignment of the given `vehicleId` to the given `tripId`. Either a variation of this event or a separate event will exist for unassigning a vehicle from a trip.
+
+The Glides application will be primarily responsible for establishing the train - trip matching. While exact details of the logic are beyond the scope of this RFC, factors that are likely to be weighed include:
+
+1. Cars entered for trips on trainsheets
+1. Additional mid-line edits to trips by Glides users, however those are represented
+1. Train movement (for instance, marking a train as no longer assigned to a trip when it reaches that trip's endpoint, or assigning it to a different trip if appropriate)
+1. Train location (for instance, marking a train as no longer assigned to a trip if its current location is no longer on the path from the start point to the end point, including being headed in the correct direction)
+1. AVI code (indicating train destination)
+
+### Required properties of Glides-produced trip assignments
+
+While there are not hard requirements around this, Glides SHOULD generally try to produce trip assignments for any vehicle tracking in the realtime data that is operating on the mainline (as opposed to in a yard). This is important for RTR to be able to make predictions without doing additional trip assignment logic (any train operating in service should be getting predictions, whether revenue train predictions in the public GTFS-rt feed or non-revenue predictions for in-station announcements about trains that won't stop). Additionally, it has benefits for operations (having a trip in the internal Glides data model provides a location to store any additional metadata, such as notes). For the time being, trip matches performed to provide a placeholder trip for a train not associated with a trip explicitly via the cars entered on a trainsheet will generally be ADDED trips that do not appear on the trainsheet interface. They will, however, generate corresponding `com.mbta.ctd.glides.trips_updated` events creating those added trips.
```

Comment:
> Glides SHOULD generally try to produce trip assignments for any vehicle tracking in the realtime data

Does this include trains that are turning around? Should we match them to the next trip even before they depart, or is it okay to leave them unmatched at the terminal (like Glides currently does)?

Reply:
I would argue that we should assign the vehicle to its next trip in that circumstance. I don't think that the GTFS-rt spec really says much about the semantics here, but I'll point out that with both bus and heavy rail, vehicles laying over at a terminal get assigned to their next trip, if known, rather than sitting around in an unknown state.